app/models/spotlight/exhibit.rb

```diff
@@ -92,19 +92,20 @@ def appearance
     Spotlight::Appearance.new(blacklight_configuration)
   end
 
-  def analytics start_date = 1.month
+  def analytics start_date = 1.month, path = nil
     return OpenStruct.new unless analytics_provider and analytics_provider.enabled?
-
-    @analytics ||= begin
-      analytics_provider.exhibit_data(self, start_date: start_date.ago)
+    @analytics ||= {}
+    @analytics[start_date] ||= begin
+      analytics_provider.exhibit_data(path || self, start_date: start_date.ago)
     end
   end
   
-  def page_analytics start_date = 1.month
+  def page_analytics start_date = 1.month, path = nil
     return [] unless analytics_provider and analytics_provider.enabled?
-
-    @page_analytics ||= begin
-      analytics_provider.page_data(self, start_date: start_date.ago)
+    
+    @page_analytics ||= {}
+    @page_analytics[start_date] ||= begin
+      analytics_provider.page_data(path || self, start_date: start_date.ago, limit: 5)
```

Comment:
Do you think we should make the limit configurable?  Perhaps a `Spotlight::Engine#config`?